lib/apiservers/engine/backends/container.go

```diff
@@ -1315,6 +1316,8 @@ func (c *Container) ContainerChanges(name string) ([]docker.Change, error) {
 			change.Kind = docker.ChangeAdd
 		case "D":
 			change.Kind = docker.ChangeDelete
+			p := strings.TrimPrefix(filepath.Base(change.Path), docker.WhiteoutPrefix)
+			change.Path = filepath.Join(filepath.Dir(change.Path), p)
```

Comment:
If you mean for the whiteout prefix, we know all changes of type delete will have it in the path as we are the ones who put it there when we build the archive :)